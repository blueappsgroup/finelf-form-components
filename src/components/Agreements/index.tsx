import React, {
  useCallback,
  useContext,
  useLayoutEffect,
  useState,
} from 'react'
import styled from 'styled-components'
import CheckboxesGroup from '../CheckboxesGroup'
import { CheckboxField } from '../'
import { FormContext } from '../../utils'
import { StyledError } from '../FormInput/base'
type AgreementType = {
  id: string
  content: string
  required?: boolean
}

type AgreementsPropTypes = {
  linksForReplace?: { [key: string]: string }
  name?: string
  groupType?: string
  requiredErorText?: string
}

const StyledErrorText = styled(StyledError)`
  position: static;
`

const Agreemnets: React.FC<AgreementsPropTypes> = ({
  linksForReplace,
  name = 'agreements',
  requiredErorText = '* Zapoznanie się z treścią regulaminu serwisu oraz polityką prywatności jest wymagane.',
}) => {
  const { id, apiUrl } = useContext(FormContext)
  const [agreements, setAgreements] = useState<AgreementType[]>([])
  const [error, setError] = useState<boolean>(false)
  const replaceLinkInAgreements = useCallback(
    (agreements: AgreementType[]): AgreementType[] => {
      const replacedAgreements = agreements.map((item) => {
        let newContent = item.content
        if (linksForReplace) {
          Object.keys(linksForReplace).forEach((key) => {
            const regxp = new RegExp(key, 'g')
            newContent = newContent.replace(regxp, linksForReplace[key])

            return newContent
          })
        }

        return { ...item, content: newContent }
      })

      return replacedAgreements
    },
    [linksForReplace]
  )
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const fetchAgreements = useCallback(async (): Promise<any> => {
    try {
      const response = await fetch(`${apiUrl}/forms/${id}/agreements`)
      const data = await response.json()
      setAgreements((linksForReplace && replaceLinkInAgreements(data)) || data)
    } catch (e) {
      console.log(e)
    }
  }, [linksForReplace, apiUrl, replaceLinkInAgreements, id])

  useLayoutEffect(() => {
    if (agreements.length === 0) {
      fetchAgreements()
    }

    if (agreements.length > 0 && !error) {
      let hasRequired = false
      agreements.some((agreement) => {
        if (agreement.required) {
          hasRequired = true
          return true
        }

        return false
      })

      if (hasRequired !== error) {
        setError(hasRequired)
      }
    }
  }, [agreements, error, fetchAgreements])

  return (
    <>
      {Array.isArray(agreements) && agreements.length > 0 && (
        <>
          <CheckboxesGroup name={name}>
            {agreements.map((item) => (
              <CheckboxField
                key={item.id}
                name={`${item.id}`}
                HTMLcontent={item.content}
                required={item.required}
              />
            ))}
          </CheckboxesGroup>
          {error && <StyledErrorText>{requiredErorText}</StyledErrorText>}
        </>
      )}
    </>
  )
}

Agreemnets.defaultProps = {
  name: 'agreements',
  groupType: 'checkboxGroup',
}

export default Agreemnets
