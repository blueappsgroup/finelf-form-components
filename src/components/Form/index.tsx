import React, { FC, ReactElement, useMemo, useState } from 'react'
import { Formik, Form } from 'formik'
import styled from 'styled-components'

import { ThemeProvider } from '../../consts/theme'
import { device } from '../../consts/sizes'
import { FormProps, FormValuesType } from '../../types'
import RedirectPage from '../RedirectPage'
import {
  FormContext,
  getFormValuesFromCache,
  resetFormValueCache,
  handleSendDataToApi,
  getFieldsValuesFromUrl,
} from '../../utils'
import { formStatuses } from '../../consts/form'

const StyledForm = styled(Form)`
  display: flex;
  flex-direction: column;
  max-width: ${(props): string => props.theme.formMaxWidth};
  justify-self: center;
  background: ${(props): string => props.theme.formBgColor};
  font-family: ${(props): string => props.theme.fontFamily};
  border-radius: 6px;
  box-shadow: ${(props): string => props.theme.formBoxShadow};
`

const FormWrapper: FC<FormProps> = ({
  children,
  onSubmit,
  customTheme,
  id,
  stepsLength,
  stepsTitles,
  hasRedirect,
  redirectUrl,
  timeToRedirect,
  redirectHeaderText,
  logoImg,
  redirectMainImg,
  redirectBgImg,
  sendDataToApi,
  apiUrl,
  transactionName,
  propertyNamesFromUrl,
  dataWithUserAgent,
}) => {
  const trasationIdValue =
    transactionName &&
    new URLSearchParams(window.location.search).get(transactionName)
  const intialValuesFromUrl = useMemo(
    () =>
      (propertyNamesFromUrl &&
        propertyNamesFromUrl.length > 0 &&
        getFieldsValuesFromUrl(propertyNamesFromUrl)) ||
      {},
    [propertyNamesFromUrl]
  )
  const [redirectUrlPath, setRedirectUrlPath] = useState(redirectUrl)
  const [initialValues, setInitialValues] = useState({
    ...intialValuesFromUrl,
    ...getFormValuesFromCache(id),
    // eslint-disable-next-line @typescript-eslint/camelcase
    trasaction_id: trasationIdValue,
  })
  const [currentStep, setCurrentStep] = useState(0)
  const [fieldsForSkip, setFieldsForSkip] = useState<string[]>([])
  const [errorFromApi, setErrorFromApi] = useState<boolean>(false)
  const shouldRedirect = !errorFromApi && hasRedirect
  const addFieldForSkip = (key: string): void =>
    setFieldsForSkip([...fieldsForSkip, key])

  const handleSubmit = async (
    values: FormValuesType,
    props: {
      resetForm: () => void
      setStatus: (status: string) => void
    }
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ): Promise<any> => {
    if (onSubmit) {
      onSubmit(values, props)
    }

    try {
      if (sendDataToApi && apiUrl) {
        const response = await handleSendDataToApi(
          values,
          apiUrl,
          id,
          fieldsForSkip,
          dataWithUserAgent
        )
        const {
          redirectUrl: urlFromApi,
          status: statusFromApi,
        } = await response.json()

        if (statusFromApi === false) {
          setErrorFromApi(true)
          return
        }

        urlFromApi && setRedirectUrlPath(urlFromApi)
      }

      props.resetForm()
      props.setStatus(formStatuses.submited)
    } catch (e) {
      props.setStatus(formStatuses.error)
    }
  }
  const handleReset = (): void => {
    resetFormValueCache(id)
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    setInitialValues({} as any)
  }

  const prevStep: Function = () => setCurrentStep(currentStep - 1)

  const nextStep: Function = () => setCurrentStep(currentStep + 1)

  return (
    <FormContext.Provider
      value={{
        id,
        apiUrl,
        stepsLength,
        currentStep,
        stepsTitleList: stepsTitles,
        nextStep,
        prevStep,
        fieldsForSkip,
        addFieldForSkip,
        errorFromApi,
      }}
    >
      <ThemeProvider customTheme={{ ...customTheme }}>
        <Formik
          enableReinitialize
          initialValues={initialValues}
          onSubmit={handleSubmit}
          onReset={handleReset}
        >
          {(props): ReactElement =>
            (shouldRedirect && props.status === formStatuses.submited && (
              <RedirectPage
                redirectUrl={redirectUrlPath}
                backgroundImage={redirectBgImg}
                logoImg={logoImg}
                headerText={redirectHeaderText}
                timeToRedirect={timeToRedirect}
                mainImg={redirectMainImg}
              />
            )) || (
              <StyledForm id={id}>
                {children}
              </StyledForm>
            )
          }
        </Formik>
      </ThemeProvider>
    </FormContext.Provider>
  )
}

export default FormWrapper
