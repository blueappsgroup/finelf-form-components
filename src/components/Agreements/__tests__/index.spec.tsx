import * as React from 'react'
import { render, act } from '@testing-library/react'

import Form from '../../Form'
import Agreements from '../'

describe('<Agreements />', () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const globalRef: any = global
  const linksForReplace = {
    RULES_URL: 'https://kredyt-dla-zadluzonych.pl/regulamin',
    PRIVACY_POLICY_URL:
      'https://kredyt-dla-zadluzonych.pl/polityka-prywatnosci/',
  }
  globalRef.fetch = jest.fn(
    () =>
      Promise.resolve({
        json: () =>
          Promise.resolve([
            {
              id: 1,
              createdOn: '2020-10-05T07:28:31.436Z',
              modifiedOn: '2020-10-05T07:28:31.436Z',
              content:
                'Oświadczam, że zapoznałem/am się z <a href="RULES_URL" title="Regulamin serwisu ">Regulaminem Serwisu</a> oraz <a href="PRIVACY_POLICY_URL" title="Polityka prywatności">Polityką Prywatności</a>. Podane przez Ciebie dane przetwarzane będą w zakresie niezbędnym do realizacji określonego żądania zgodnie z art. 6 ust. 1 lit. b) RODO tj. do podjęcia działań przed zawarciem umowy, podejmowania działań w związku z jej wykonywaniem lub na podstawie art. 6 ust. 1 lit. f) RODO tj. w celu realizacji uzasadnionego interesu administratora. Szczegóły w Polityce Prywatności.',
              required: true,
            },
            {
              id: 2,
              createdOn: '2020-10-05T07:28:39.462Z',
              modifiedOn: '2020-10-05T07:28:39.462Z',
              content:
                'Wyrażam zgodę na otrzymywanie od Usługodawcy tj. Da Vinci Investment sp. z o.o. z siedzibą w Bydgoszczy przy ul. Krzysztofa Gotowskiego 6, 85-030 Bydgoszcz (numer KRS: 0000728065) na podane przeze mnie dane kontaktowe informacji handlowych i marketingu bezpośredniego za pomocą poczty elektronicznej, wiadomości sms/mms i połączeń telefonicznych w celu przedstawienia dedykowanych dla mnie ofert lub usług Usługodawcy. Zgoda obejmuje także profilowanie danych, które ma na celu dopasowanie oferty do moich oczekiwań. Wyrażenie zgody jest dobrowolne. Wiem, że mam prawo do cofnięcia zgody w dowolnym momencie. Przyjmuję do wiadomości, że wyrażenie powyższej zgody oznacza jednocześnie zgodę na przetwarzanie przez Usługodawcę moich danych osobowych w celu promocji oraz marketingu bezpośredniego własnych produktów i usług.',
              required: false,
            },
            {
              id: 3,
              createdOn: '2020-10-05T07:28:48.344Z',
              modifiedOn: '2020-10-05T07:28:48.344Z',
              content:
                'Wyrażam zgodę na otrzymywanie od partnerów serwisu SITE_NAME (wymienionych na stronie internetowej w linku: PARTNERS_URL) na podane przeze mnie dane kontaktowe informacji handlowych i marketingu bezpośredniego za pomocą poczty elektronicznej, wiadomości sms/mms i połączeń telefonicznych w celu przedstawienia przez nich dedykowanych dla mnie ofert lub usług, związanych z produktami pożyczkowymi. Zgoda obejmuje także profilowanie danych, które ma na celu dopasowanie oferty do moich oczekiwań. Wyrażenie zgody jest dobrowolne. Wiem, że mam prawo do cofnięcia zgody w dowolnym momencie. Przyjmuję do wiadomości, że wyrażenie powyższej zgody oznacza jednocześnie zgodę na przetwarzanie przez powyższych partnerów moich danych osobowych w celu promocji oraz marketingu bezpośredniego ich produktów i usług.',
              required: false,
            },
          ]),
      })
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ) as any

  beforeEach(() => {
    globalRef.fetch.mockClear()
  })
  const onSubmit = jest.fn()
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const setupWrapper: (config: any) => any = async ({
    formId = 'testForm',
    formProps,
    ...rest
  }) => {
    let wrapper

    await act(
      async () =>
        (wrapper = render(
          <Form id={formId} onSubmit={onSubmit} {...formProps}>
            <Agreements {...rest} />
          </Form>
        ))
    )

    return wrapper
  }

  it('matches snapshot', async () => {
    const { container } = await setupWrapper({})

    expect(container).toMatchSnapshot()
  })

  it('to has proper count of checkoxes', async () => {
    const { baseElement } = await setupWrapper({})
    const checkboxesCount = baseElement.querySelectorAll('input').length

    expect(checkboxesCount).toBe(4)
  })

  it('select all checkbox should select all checkboxes', async () => {
    let allCheckboxesSelected = true
    const { baseElement } = await setupWrapper({})
    const checkboxes = baseElement.querySelectorAll('input')

    await act(async () => {
      checkboxes[0].click()
    })

    checkboxes.forEach((checkbox) => {
      if (checkbox.value === 'false') {
        allCheckboxesSelected = false
      }
    })

    expect(allCheckboxesSelected).toBeTruthy()
  })

  it('unselect all checkbox should unselect all checkboxes', async () => {
    let allCheckboxesSelected = true
    const { baseElement } = await setupWrapper({})
    const checkboxes = baseElement.querySelectorAll('input')

    await act(async () => {
      checkboxes[0].click()
    })

    expect(allCheckboxesSelected).toBeTruthy()

    await act(async () => {
      checkboxes[0].click()
    })

    checkboxes.forEach((checkbox) => {
      if (checkbox.value === 'false') {
        allCheckboxesSelected = false
      }
    })

    expect(allCheckboxesSelected).toBeFalsy()
  })

  it('will agreements link replace', async () => {
    const { baseElement, getByText } = await setupWrapper({ linksForReplace })
    expect(getByText('Regulaminem Serwisu').closest('a')).toHaveAttribute(
      'href',
      linksForReplace.RULES_URL
    )
    expect(getByText('Polityką Prywatności').closest('a')).toHaveAttribute(
      'href',
      linksForReplace.PRIVACY_POLICY_URL
    )

    expect(baseElement).toMatchSnapshot()
  })
})
