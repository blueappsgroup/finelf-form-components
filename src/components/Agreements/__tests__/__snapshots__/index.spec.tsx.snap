// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Agreements /> matches snapshot 1`] = `
<div>
  <form
    action="#"
    class="sc-fubCfw clDFWv"
    id="testForm"
  >
    <div
      class="sc-pFZIQ hlPOqE"
    >
      <label
        class="sc-kstrdz irJNIW"
      >
        <div
          class="sc-kEjbxe kzwsDs"
        >
          <input
            class="sc-crrsfI ljDaNc"
            id="agreements.selectAll"
            label="Zaznacz wszystkie zgody"
            name="agreements.selectAll"
            type="checkbox"
            value="false"
          />
          <div
            class="sc-dQppl fMeTOd"
          >
            <svg
              class="sc-iqHYGH dFoDCi"
              viewBox="0 0 24 24"
            >
              <polyline
                points="20 6 9 17 4 12"
              />
            </svg>
          </div>
        </div>
        <span
          class="sc-bqyKva hFXhsJ"
        >
          Zaznacz wszystkie zgody
        </span>
        <div
          class="sc-fodVxV fFWVU"
        />
      </label>
      <span
        class="sc-jrAGrp ebhOan"
      />
    </div>
    <div
      class="sc-pFZIQ kFwajQ"
    >
      <label
        class="sc-kstrdz irJNIW"
      >
        <div
          class="sc-kEjbxe kzwsDs"
        >
          <input
            class="sc-crrsfI ljDaNc"
            id="agreements.1"
            name="agreements.1"
            required=""
            type="checkbox"
            value="false"
          />
          <div
            class="sc-dQppl gPlrmW"
            required=""
          >
            <svg
              class="sc-iqHYGH dFoDCi"
              viewBox="0 0 24 24"
            >
              <polyline
                points="20 6 9 17 4 12"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-fodVxV fFWVU"
        >
          <span
            class="sc-bqyKva hFXhsJ"
          >
            Oświadczam, że zapoznałem/am się z 
            <a
              href="RULES_URL"
              title="Regulamin serwisu "
            >
              Regulaminem Serwisu
            </a>
             oraz 
            <a
              href="PRIVACY_POLICY_URL"
              title="Polityka prywatności"
            >
              Polityką Prywatności
            </a>
            . Podane przez Ciebie dane przetwarzane będą w zakresie niezbędnym do realizacji określonego żądania zgodnie z art. 6 ust. 1 lit. b) RODO tj. do podjęcia działań przed zawarciem umowy, podejmowania działań w związku z jej wykonywaniem lub na podstawie art. 6 ust. 1 lit. f) RODO tj. w celu realizacji uzasadnionego interesu administratora. Szczegóły w Polityce Prywatności.
          </span>
        </div>
      </label>
      <span
        class="sc-jrAGrp ebhOan"
      />
    </div>
    <div
      class="sc-pFZIQ kFwajQ"
    >
      <label
        class="sc-kstrdz irJNIW"
      >
        <div
          class="sc-kEjbxe kzwsDs"
        >
          <input
            class="sc-crrsfI ljDaNc"
            id="agreements.2"
            name="agreements.2"
            type="checkbox"
            value="false"
          />
          <div
            class="sc-dQppl fMeTOd"
          >
            <svg
              class="sc-iqHYGH dFoDCi"
              viewBox="0 0 24 24"
            >
              <polyline
                points="20 6 9 17 4 12"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-fodVxV fFWVU"
        >
          <span
            class="sc-bqyKva hFXhsJ"
          >
            Wyrażam zgodę na otrzymywanie od Usługodawcy tj. Da Vinci Investment sp. z o.o. z siedzibą w Bydgoszczy przy ul. Krzysztofa Gotowskiego 6, 85-030 Bydgoszcz (numer KRS: 0000728065) na podane przeze mnie dane kontaktowe informacji handlowych i marketingu bezpośredniego za pomocą poczty elektronicznej, wiadomości sms/mms i połączeń telefonicznych w celu przedstawienia dedykowanych dla mnie ofert lub usług Usługodawcy. Zgoda obejmuje także profilowanie danych, które ma na celu dopasowanie oferty do moich oczekiwań. Wyrażenie zgody jest dobrowolne. Wiem, że mam prawo do cofnięcia zgody w dowolnym momencie. Przyjmuję do wiadomości, że wyrażenie powyższej zgody oznacza jednocześnie zgodę na przetwarzanie przez Usługodawcę moich danych osobowych w celu promocji oraz marketingu bezpośredniego własnych produktów i usług.
          </span>
        </div>
      </label>
      <span
        class="sc-jrAGrp ebhOan"
      />
    </div>
    <div
      class="sc-pFZIQ kFwajQ"
    >
      <label
        class="sc-kstrdz irJNIW"
      >
        <div
          class="sc-kEjbxe kzwsDs"
        >
          <input
            class="sc-crrsfI ljDaNc"
            id="agreements.3"
            name="agreements.3"
            type="checkbox"
            value="false"
          />
          <div
            class="sc-dQppl fMeTOd"
          >
            <svg
              class="sc-iqHYGH dFoDCi"
              viewBox="0 0 24 24"
            >
              <polyline
                points="20 6 9 17 4 12"
              />
            </svg>
          </div>
        </div>
        <div
          class="sc-fodVxV fFWVU"
        >
          <span
            class="sc-bqyKva hFXhsJ"
          >
            Wyrażam zgodę na otrzymywanie od partnerów serwisu SITE_NAME (wymienionych na stronie internetowej w linku: PARTNERS_URL) na podane przeze mnie dane kontaktowe informacji handlowych i marketingu bezpośredniego za pomocą poczty elektronicznej, wiadomości sms/mms i połączeń telefonicznych w celu przedstawienia przez nich dedykowanych dla mnie ofert lub usług, związanych z produktami pożyczkowymi. Zgoda obejmuje także profilowanie danych, które ma na celu dopasowanie oferty do moich oczekiwań. Wyrażenie zgody jest dobrowolne. Wiem, że mam prawo do cofnięcia zgody w dowolnym momencie. Przyjmuję do wiadomości, że wyrażenie powyższej zgody oznacza jednocześnie zgodę na przetwarzanie przez powyższych partnerów moich danych osobowych w celu promocji oraz marketingu bezpośredniego ich produktów i usług.
          </span>
        </div>
      </label>
      <span
        class="sc-jrAGrp ebhOan"
      />
    </div>
    <span
      class="sc-bBXqnf sc-cOajty dTtkZj leWBTa"
    >
      * Zapoznanie się z treścią regulaminu serwisu oraz polityką prywatności jest wymagane.
    </span>
    <button
      class="sc-iBPRYJ hcYpVh"
      disabled=""
      type="submit"
    >
      Wyślij
    </button>
  </form>
</div>
`;

exports[`<Agreements /> will agreements link replace 1`] = `
<body>
  <div>
    <form
      action="#"
      class="sc-fubCfw clDFWv"
      id="testForm"
    >
      <div
        class="sc-pFZIQ hlPOqE"
      >
        <label
          class="sc-kstrdz irJNIW"
        >
          <div
            class="sc-kEjbxe kzwsDs"
          >
            <input
              class="sc-crrsfI ljDaNc"
              id="agreements.selectAll"
              label="Zaznacz wszystkie zgody"
              name="agreements.selectAll"
              type="checkbox"
              value="false"
            />
            <div
              class="sc-dQppl fMeTOd"
            >
              <svg
                class="sc-iqHYGH dFoDCi"
                viewBox="0 0 24 24"
              >
                <polyline
                  points="20 6 9 17 4 12"
                />
              </svg>
            </div>
          </div>
          <span
            class="sc-bqyKva hFXhsJ"
          >
            Zaznacz wszystkie zgody
          </span>
          <div
            class="sc-fodVxV fFWVU"
          />
        </label>
        <span
          class="sc-jrAGrp ebhOan"
        />
      </div>
      <div
        class="sc-pFZIQ kFwajQ"
      >
        <label
          class="sc-kstrdz irJNIW"
        >
          <div
            class="sc-kEjbxe kzwsDs"
          >
            <input
              class="sc-crrsfI ljDaNc"
              id="agreements.1"
              name="agreements.1"
              required=""
              type="checkbox"
              value="false"
            />
            <div
              class="sc-dQppl gPlrmW"
              required=""
            >
              <svg
                class="sc-iqHYGH dFoDCi"
                viewBox="0 0 24 24"
              >
                <polyline
                  points="20 6 9 17 4 12"
                />
              </svg>
            </div>
          </div>
          <div
            class="sc-fodVxV fFWVU"
          >
            <span
              class="sc-bqyKva hFXhsJ"
            >
              Oświadczam, że zapoznałem/am się z 
              <a
                href="https://kredyt-dla-zadluzonych.pl/regulamin"
                title="Regulamin serwisu "
              >
                Regulaminem Serwisu
              </a>
               oraz 
              <a
                href="https://kredyt-dla-zadluzonych.pl/polityka-prywatnosci/"
                title="Polityka prywatności"
              >
                Polityką Prywatności
              </a>
              . Podane przez Ciebie dane przetwarzane będą w zakresie niezbędnym do realizacji określonego żądania zgodnie z art. 6 ust. 1 lit. b) RODO tj. do podjęcia działań przed zawarciem umowy, podejmowania działań w związku z jej wykonywaniem lub na podstawie art. 6 ust. 1 lit. f) RODO tj. w celu realizacji uzasadnionego interesu administratora. Szczegóły w Polityce Prywatności.
            </span>
          </div>
        </label>
        <span
          class="sc-jrAGrp ebhOan"
        />
      </div>
      <div
        class="sc-pFZIQ kFwajQ"
      >
        <label
          class="sc-kstrdz irJNIW"
        >
          <div
            class="sc-kEjbxe kzwsDs"
          >
            <input
              class="sc-crrsfI ljDaNc"
              id="agreements.2"
              name="agreements.2"
              type="checkbox"
              value="false"
            />
            <div
              class="sc-dQppl fMeTOd"
            >
              <svg
                class="sc-iqHYGH dFoDCi"
                viewBox="0 0 24 24"
              >
                <polyline
                  points="20 6 9 17 4 12"
                />
              </svg>
            </div>
          </div>
          <div
            class="sc-fodVxV fFWVU"
          >
            <span
              class="sc-bqyKva hFXhsJ"
            >
              Wyrażam zgodę na otrzymywanie od Usługodawcy tj. Da Vinci Investment sp. z o.o. z siedzibą w Bydgoszczy przy ul. Krzysztofa Gotowskiego 6, 85-030 Bydgoszcz (numer KRS: 0000728065) na podane przeze mnie dane kontaktowe informacji handlowych i marketingu bezpośredniego za pomocą poczty elektronicznej, wiadomości sms/mms i połączeń telefonicznych w celu przedstawienia dedykowanych dla mnie ofert lub usług Usługodawcy. Zgoda obejmuje także profilowanie danych, które ma na celu dopasowanie oferty do moich oczekiwań. Wyrażenie zgody jest dobrowolne. Wiem, że mam prawo do cofnięcia zgody w dowolnym momencie. Przyjmuję do wiadomości, że wyrażenie powyższej zgody oznacza jednocześnie zgodę na przetwarzanie przez Usługodawcę moich danych osobowych w celu promocji oraz marketingu bezpośredniego własnych produktów i usług.
            </span>
          </div>
        </label>
        <span
          class="sc-jrAGrp ebhOan"
        />
      </div>
      <div
        class="sc-pFZIQ kFwajQ"
      >
        <label
          class="sc-kstrdz irJNIW"
        >
          <div
            class="sc-kEjbxe kzwsDs"
          >
            <input
              class="sc-crrsfI ljDaNc"
              id="agreements.3"
              name="agreements.3"
              type="checkbox"
              value="false"
            />
            <div
              class="sc-dQppl fMeTOd"
            >
              <svg
                class="sc-iqHYGH dFoDCi"
                viewBox="0 0 24 24"
              >
                <polyline
                  points="20 6 9 17 4 12"
                />
              </svg>
            </div>
          </div>
          <div
            class="sc-fodVxV fFWVU"
          >
            <span
              class="sc-bqyKva hFXhsJ"
            >
              Wyrażam zgodę na otrzymywanie od partnerów serwisu SITE_NAME (wymienionych na stronie internetowej w linku: PARTNERS_URL) na podane przeze mnie dane kontaktowe informacji handlowych i marketingu bezpośredniego za pomocą poczty elektronicznej, wiadomości sms/mms i połączeń telefonicznych w celu przedstawienia przez nich dedykowanych dla mnie ofert lub usług, związanych z produktami pożyczkowymi. Zgoda obejmuje także profilowanie danych, które ma na celu dopasowanie oferty do moich oczekiwań. Wyrażenie zgody jest dobrowolne. Wiem, że mam prawo do cofnięcia zgody w dowolnym momencie. Przyjmuję do wiadomości, że wyrażenie powyższej zgody oznacza jednocześnie zgodę na przetwarzanie przez powyższych partnerów moich danych osobowych w celu promocji oraz marketingu bezpośredniego ich produktów i usług.
            </span>
          </div>
        </label>
        <span
          class="sc-jrAGrp ebhOan"
        />
      </div>
      <span
        class="sc-bBXqnf sc-cOajty dTtkZj leWBTa"
      >
        * Zapoznanie się z treścią regulaminu serwisu oraz polityką prywatności jest wymagane.
      </span>
      <button
        class="sc-iBPRYJ hcYpVh"
        disabled=""
        type="submit"
      >
        Wyślij
      </button>
    </form>
  </div>
</body>
`;
